spec.bs

```diff
@@ -759,6 +786,185 @@ in the [[#nested-traversables-intro]].
+    If these steps were invoked by user gesture, [=focusing steps=] can be called with
+    [=focus-unfenced|unfenced=] set to true. For elements that are
+    [=form-associated custom elements=], user agents should use their [=face validation anchor=]
+    instead, for the purposes of these actions.
+</div>
+
+<div algorithm=has-focus-steps>
+  Add a step after step 2 of the while loop in the [=has focus steps=] algorithm that reads:
+
+  3. If the [=focused area=] of |candidate| is a [=fenced navigable container=] with a non-null
+    [=fenced navigable container/fenced navigable=], then set |candidate| to the [=navigable/active
+    document=] of that [=fenced navigable container=]'s [=fenced navigable container/fenced
+    navigable=].
+</div>
+
+<div algorithm=focus-chain>
+  Modify step 3 of the while loop in the [=focus chain=] algorithm to read:
+  3. If |currentObject| is a [=focusable area=], then set |currentObject| to |currentObject|'s [=DOM
+     anchor=]'s [=Node/node document=].
+
+     Otherwise, if |currentObject| is a {{Document}} whose [=node navigable=]'s [=navigable/parent=]
+     is non-null, then set |currentObject| to |currentObject|'s [=node navigable=]'s
+     [=navigable/parent=].
+
+     Otherwise, if |currentObject| is a {{Document}} whose [=node navigable=] is a [=traversable
+     navigable=] whose [=traversable navigable/unfenced parent=] is non-null, then set
+     |currentObject| to |currentObject|'s [=node navigable=]'s [=traversable navigable/unfenced
+     parent=].
+
+     Otherwise, [=iteration/break=].
+</div>
+  
+<div algorithm=get-the-focusable-area>
+  Modify the [=get the focusable area=] algorithm. Add a new case to the switch statement:
+
+  <dl class="switch">
+    <dt>If <var ignore>focus target</var> is a [=fenced navigable container=] with a non-null [=fenced navigable container/fenced navigable=]</dt>
+
+    <dd><p>Return the [=fenced navigable container=]'s [=fenced navigable container/fenced
+    navigable=]'s [=navigable/active document=].</p></dd>
+  </dl>
+
+  Note: This algorithm can unconditionally "jump the fence" boundary because its return value always
+  feeds into an algorithm that _does_ carefully consider the fence boundary.
+</div>
+
+<div algorithm=currently-focused-area-of-a-top-level-traversable>
+  Modify step 3 of the [=currently focused area of a top-level traversable=] algorithm to read:
+
+  3. While |candidate|'s [=focused area=] is either a [=navigable container=] with a non-null
+    [=navigable container/content navigable=] or a [=fenced navigable container=] with a non-null
+    [=fenced navigable container/fenced navigable=]: set |candidate| to the
+    [=navigable/active document=] of either that [=navigable container=]'s
+    [=navigable container/content navigable=] or that [=fenced navigable container=]'s
+    [=fenced navigable container/fenced navigable=], whichever is non-null.
+</div>
+
+<div algorithm=sequential-focus-navigation-patch>
```

Comment:
All of the sequential focus and navigation changes are unfenced by default it seems. I just want to double check: that's because there are only initiated with user actions, right? I think I'm going to land this right not but please respond here after just for posterity

Reply:
That is correct.

https://docs.google.com/spreadsheets/d/1p7_LtfkB1BAFx_Iu_vQ9J0zLHgd2SaC5ih4ANfhYp0E/edit#gid=533819146&range=152:152